metricbeat/mb/parse/url.go

```diff
@@ -208,18 +208,27 @@ func getURL(rawURL, scheme, username, password, path, query string) (*url.URL, e
 		u.Path = path
 	}
 
-	// Add the query params to existing query parameters overwriting any
-	// keys that already exist.
+	//Adds the query params in the url
+	u, err = SetQueryParams(u, query)
+	return u, err
+}
+
+// SetQueryParams adds the query params to existing query parameters overwriting any
+// keys that already exist.
+func SetQueryParams(u *url.URL, query string) (*url.URL, error) {
```

Comment:
@sayden do you think #8292 could help here?

Reply:
Yes definitely. After our discussion we decided it was a bit more clear to have the params in the YAML instead of as an "ugly" string. It's one step far from merge so I can push it forward a bit more today.

Reply:
@narph We are adding query params to the YAML definition as you can see here https://github.com/elastic/beats/pull/8292 so you won't need to parse the params from the URL giving a clearer configuration file. It will be as simple as adding a map to the YAML with the desired params as you can see in this example https://github.com/elastic/beats/pull/8292#issuecomment-421350068

Reply:
@sayden , your PR comes in handy indeed in this case.
@jsoriano, as per our conversation I will go for an event per process approach and move this setup to a new metricset.